_sandboxbuildboxrun.py: Restore terminal after exit of interactive child

abderrahim opened the pull request:
This is a port of 8f401127c450ea1c2a1e2878d2ac28a7166d355c to the buildbox-run sandbox

Thanks to @adoakley for investigating the issue. This was previously #41

Fixes #1690

abderrahim committed:
_sandboxbuildboxrun.py: Restore terminal after exit of interactive child

This is a port of 8f401127c450ea1c2a1e2878d2ac28a7166d355c to the
buildbox-run sandbox

Fixes #1690

abderrahim commented:
@juergbi I guess you're the best placed to review as the author of the original commit.

abderrahim commented:
Seems a bit more complicated than this: there are a few test failures with `Inappropriate ioctl for device`. I wonder why, since I don't think we have tests that run bst interactively.

cs-shadow commented:
the logs have disappeared now so it's difficult to say now but I just wanted to respond to this bit:

> Seems a bit more complicated than this: there are a few test failures with Inappropriate ioctl for device. I wonder why, since I don't think we have tests that run bst interactively.

We actually do! I was very happy to add the first batch of them in https://gitlab.com/BuildStream/buildstream/-/merge_requests/1706 :)

it's possible `pexpect` didn't like deal well with this change.

abderrahim committed:
_sandboxbuildboxrun: don't try to restore terminal if not on a terminal

This happens in our tests where we don't have a terminal attached even
for interactive operations

abderrahim commented:
I pushed a fix to also check that stdin is a terminal before calling ioctl.

juergbi reviewed (approved):
LGTM and seems to work fine in a manual test.
